File: Healthcare-app/ios/MDBRealmPatient/MDBRealmPatient/Appointment.swift
//
//  Appointment.swift
//  MDBRealmPatient
//
//  Created by Mackbook on 07/05/22.
//

import Foundation
import RealmSwift

class Appointment: Object {
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var end: Date?
    @Persisted var identifier: String?
    @Persisted var nurseIdentifier: String?
    @Persisted var participant: List<Appointment_Participant>
    @Persisted var patientIdentifier: String?
    @Persisted var patientInstruction: String?
    @Persisted var practitionerIdentifier: String?
    @Persisted var reasonReference: Reference?
    @Persisted var slot: Int?
    @Persisted var start: Date?
    @Persisted var status: String?
}
class Appointment_Participant: EmbeddedObject {
    @Persisted var actor: Reference?
    @Persisted var required: Bool?
    @Persisted var status: String?
}

File: Healthcare-app/android/MDBRealmDoctor/app/src/main/java/com/wekanmdb/storeinventory/model/appoinment/Appointment.kt
package com.wekanmdb.storeinventory.model.appoinment

import com.wekanmdb.storeinventory.model.embededclass.Appointment_Participant
import com.wekanmdb.storeinventory.model.embededclass.Reference
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import org.bson.types.ObjectId
import java.util.*

open class Appointment (
    @PrimaryKey var _id: ObjectId = ObjectId(),
    var identifier: String? = null,
    var status: String? = null,
    var reasonReference: Reference? = null,
    var start: Date?=null,
    var end: Date?=null,
    var patientInstruction: String? = null,
    var participant: RealmList<Appointment_Participant> = RealmList(),
    var slot: Long? = null,
    var patientIdentifier: String? = null,
    var practitionerIdentifier: String? = null,
    var nurseIdentifier: String? = null
) : RealmObject() {}